### PR TITLE
Public Files: Downloading using Downloader.download

### PR DESCRIPTION
**Pull request by mmjee**
# Pull Request Template

Resolves [AR-4063](https://team-1624093970686.atlassian.net/browse/AR-4063).

## Blocking dependencies

- [AR-4084](https://team-1624093970686.atlassian.net/browse/AR-4084)

## Changes

See the ticket.

## Checklist

- [x] The branch name follows the format: `developer/AR-XXX-issue-name`.
- [x] The changes have been tested locally.

### CODE REVIEW

**Review by SauravKanchan — COMMENTED** [review]
@mmjee  Can you update usage.MD?

**Comment by mmjee** [comment]
On 9/22/22 10:06, Saurav Kanchan wrote:
> @SauravKanchan commented on this pull request.
> 
> @mmjee  Can you update usage.MD?
> 
> 
> 
Done.
-- 
Regards,

File: src/Uploader.ts [hunk 1]
```diff
@@ -183,17 +215,19 @@ export class Uploader {
...
-            length: 64,
-            name: 'AES-CTR',
-          },
-          key,
-          chunk,
-        );
+        if (!params.publicFile) {
+          chunk = await window.crypto.subtle.encrypt(
+            {
+              counter: generateCounterFromPartNumber(uploadedParts),
```
Comment (SauravKanchan) [inline src/Uploader.ts:223]: this should be counter

File: src/Downloader.ts [hunk 2]
```diff
@@ -65,69 +66,106 @@ export class Downloader {
...
+            address: this.appAddress
+          }
+        })
+
+        const u = new URL(storageHost);
+        u.pathname = '/api/v2/file/public/' + this.appAddress + '/' + did
+
+        let downloaded = 0
+        for (let i = 0; i < fileMeta.size; i += chunkSize) {
+          const range = `bytes=${i}-${Math.min(i + chunkSize, fileMeta.size)}`;
```
Comment (SauravKanchan) [inline src/Downloader.ts:92]: This should be
```js
const range = `bytes=${i}-${Math.min(i + chunkSize, fileMeta.size)-1}`;
```

**Review by SauravKanchan — CHANGES_REQUESTED** [review]

**Pull request merged**